### PR TITLE
feat: add podLabels parameter

### LINKED ISSUE #712: [vpa] add podLabels to VPA chart
**Is your feature request related to a problem? Please describe.**
Need a way to add custom labels to the pods

**Describe the solution you'd like**
Add podLabels to values.yaml

**Describe alternatives you've considered**
N/A

**Additional context**
Need a way to set custom labels on pods through Helm chart

### PR DESCRIPTION
**Pull request by francRang**
**Why This PR?**
Need a way to add custom labels to all pods and/or specific component pods

Fixes #712

**Changes**
Changes proposed in this pull request:

* podLabels added to values.yaml
* recommender.podLabels added to values.yaml
* updater.podLabels added to values.yaml
* admissionController.podLabels added to values.yaml

**Checklist:**

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] Any new values have been added to the README for the Chart, or `helm-docs --sort-values-order=file` has been run for the charts that support it.

### CODE REVIEW

File: stable/vpa/templates/_helpers.tpl [hunk 1]
```diff
@@ -40,6 +40,18 @@ helm.sh/chart: {{ include "vpa.chart" . }}
...
+{{- end }}
+{{- if .Values.recommender.podLabels }}
+{{ toYaml .Values.recommender.podLabels }}
+{{- end }}
+{{- if .Values.updater.podLabels }}
+{{ toYaml .Values.updater.podLabels }}
+{{- end }}
+{{- if .Values.admissionController.podLabels }}
+{{ toYaml .Values.admissionController.podLabels }}
+{{- end }}
```
Comment (sudermanjr) [inline stable/vpa/templates/_helpers.tpl]: If I set .Values.updater.podLabels, won't they end up on every deployment with this setup? I think the component-specific ones need to be added to the individual deployment templates

**Review by sudermanjr — CHANGES_REQUESTED** [review]

File: stable/vpa/templates/_helpers.tpl (hunk 1 above)
Comment (francRang) [reply stable/vpa/templates/_helpers.tpl]: Thanks for the catch! Pushed an update :)

**Comment by sudermanjr** [comment]
Another thing that might be nice is to add some test labels to the `ci/test-values.yaml` file to make sure this gets covered in the e2e tests.

File: stable/vpa/templates/admission-controller-deployment.yaml [hunk 2]
```diff
@@ -18,7 +18,10 @@ spec:
       annotations:
         {{- toYaml . | nindent 8 }}
     {{- end }}
+    {{- with .Values.admissionController.podLabels }}
       labels:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
```
Comment (sudermanjr) [inline stable/vpa/templates/admission-controller-deployment.yaml]:
```suggestion
      labels:
      {{- with .Values.admissionController.podLabels }}
        {{- toYaml . | nindent 8 }}
      {{- end }}
```

File: stable/vpa/templates/recommender-deployment.yaml [hunk 3]
```diff
@@ -18,7 +18,10 @@ spec:
       annotations:
         {{- toYaml . | nindent 8 }}
     {{- end }}
+    {{- with .Values.recommender.podLabels }}
       labels:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
```
Comment (sudermanjr) [inline stable/vpa/templates/recommender-deployment.yaml]:
```suggestion
      labels:
      {{- with .Values.recommender.podLabels }}
        {{- toYaml . | nindent 8 }}
      {{- end }}
```

File: stable/vpa/templates/updater-deployment.yaml [hunk 4]
```diff
@@ -18,7 +18,10 @@ spec:
       annotations:
         {{- toYaml . | nindent 8 }}
     {{- end }}
+    {{- with .Values.updater.podLabels }}
       labels:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
```
Comment (sudermanjr) [inline stable/vpa/templates/updater-deployment.yaml]:
```suggestion
      labels:
      {{- with .Values.updater.podLabels }}
        {{- toYaml . | nindent 8 }}
      {{- end }}
```

**Review by sudermanjr — CHANGES_REQUESTED** [review]
I think the `{{- with` needs to be inside labels since there's additional labels. This is likely what's causing the e2e errors.

File: stable/vpa/templates/admission-controller-deployment.yaml (hunk 2 above)
Comment (francRang) [reply stable/vpa/templates/admission-controller-deployment.yaml]: Submitted requested changes

File: stable/vpa/templates/updater-deployment.yaml (hunk 4 above)
Comment (francRang) [reply stable/vpa/templates/updater-deployment.yaml]: This has been addressed. Submitted requested changes

File: stable/vpa/templates/recommender-deployment.yaml (hunk 3 above)
Comment (francRang) [reply stable/vpa/templates/recommender-deployment.yaml]: Submitted requested changes

**Comment by sudermanjr** [comment]
Looks great! Now if you can please sign the CLA we'll be good to go.

**Comment by fr4nkR** [comment]
> Looks great! Now if you can please sign the CLA we'll be good to go.

For whatever reason the CLA assistant doesn't mark it as signed, even though I already signed and have clicked on the "recheck" option multiple times.

From what you recall, has anyone ever experienced this issue before?

**Comment by francRang** [comment]
> I think the `{{- with` needs to be inside labels since there's additional labels. This is likely what's causing the e2e errors.



> Another thing that might be nice is to add some test labels to the `ci/test-values.yaml` file to make sure this gets covered in the e2e tests.

This has been resolved

**Comment by francRang** [comment]
> I think the `{{- with` needs to be inside labels since there's additional labels. This is likely what's causing the e2e errors.
Can you please mark this as addressed so we can merge?

**Review by sudermanjr — APPROVED** [review]
Awesome! Thanks so much!

**Pull request merged**